services/blockchain-connector/shared/sdk/client.js

```diff
@@ -32,24 +32,56 @@ const RETRY_INTERVAL = config.apiClient.instantiation.retryInterval;
 const MAX_INSTANTIATION_WAIT_TIME = config.apiClient.instantiation.maxWaitTime;
 const NUM_REQUEST_RETRIES = config.apiClient.request.maxRetries;
 const ENDPOINT_INVOKE_RETRY_DELAY = config.apiClient.request.retryDelay;
+const CLIENT_ALIVE_ASSUMPTION_TIME = config.apiClient.aliveAssumptionTime;
 
 // Caching and flags
 let clientCache;
 let instantiationBeginTime;
+let lastClientAliveTime;
 let isInstantiating = false;
+let isClientAlive = false;
 
-const checkIsClientAlive = () =>
-	clientCache && clientCache._channel && clientCache._channel.isAlive;
+const pongListener = res => {
+	isClientAlive = true;
+	lastClientAliveTime = Date.now();
+	return res(true);
+};
+
+const checkIsClientAlive = async () =>
+	// eslint-disable-next-line consistent-return
+	new Promise(resolve => {
+		if (!clientCache || (clientCache._channel && !clientCache._channel.isAlive)) {
+			return resolve(false);
+		}
+
+		if (
+			config.isUseLiskIPCClient ||
+			Date.now() - lastClientAliveTime > CLIENT_ALIVE_ASSUMPTION_TIME
+		) {
+			return resolve(clientCache._channel && clientCache._channel.isAlive);
+		}
+
+		const boundPongListener = () => pongListener(resolve);
+
+		clientCache._channel._ws.on('pong', boundPongListener);
+		clientCache._channel._ws.ping(() => {});
+
+		// eslint-disable-next-line consistent-return
+		setTimeout(() => {
+			clientCache._channel._ws.removeEventListener('pong', boundPongListener);
+			if (!isClientAlive) return resolve(false);
+		}, RETRY_INTERVAL);
```

Comment:
we can have this information readily available by sending a ping every 5 seconds. This would be helpful as we (almost) always invoke `checkIsClientAlive` at least once every 5 seconds timeframe.